com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ServiceMetadataBuilder.java

```diff
@@ -0,0 +1,56 @@
+package com.sap.cloud.lm.sl.cf.core.cf.v2;
+
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.processor.ServiceMtaMetadataExtractor;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadataLabels;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaResource;
+import com.sap.cloud.lm.sl.cf.core.util.NameUtil;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.Module;
+import com.sap.cloud.lm.sl.mta.model.Resource;
+import org.cloudfoundry.client.v3.Metadata;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ServiceMetadataBuilder {
```

Comment:
This class has a lot in common with the application metadata builder.
Is it possible that some form of generalization be done to reduce duplication?

Reply:
Check my above comment and you will find a way to workaround it.